### PR TITLE
fix(List): update List usages

### PR DESCRIPTION
**Pull request by fabienhebert**
## Type

- Enhancement

### Summarise concisely:

Update some list usages

#### The changes of this proposal:

**Selectable rows :**
- Remove the <List> properties :  `idKey` and `data` which was use only for selecting allRow, instead each row selectable row register to the context
- Remove the <List> property `isSelectable`, to know if we can select row, instead we detect if props `selectedIds` and `onSelectedIdsChange` are provided or not
- Select/Unselect all is now disabled if there is no selectable rows

**Expandable rows :**
- No more List.Expandable element, but a a prop `expandable`. It helps identify row cells content with expandable content. It also allow to remove the `isExpandable` prop on `List.Row`
- Bug fix: if a row can be extended by clicking, a column is created in header to fix headerCell / bodyCell alignment
- Bug fix: the last cell wasn't properly identified some type, so the last cell available space was sometime too high
- Remove `hideArrow`, it is now implied if `isExpanded` (manual expanding) is not undefined
- Force `isHoverable` if a row can be expanded

**Column sorting**
- onClick prop `column` has been renamed `columnId` since it refers to prop `id` provided to a column.
- onClick prop `order` is now `newOrder` and return the opposite value of the `sort` value provided to a column

**Others**
- The border/separator between cells and expandable is now the same than row border (if you specify a sentiment)
- Prop `variant` has been renamed `sentiment`
- Updates `stories` to give more info about some props

### CODE REVIEW

**Review by matthprost — CHANGES_REQUESTED** [review]
Screenshot 2023-02-08 at 11 30 00

When `selectedIds={[]}` it leave global checkbox checked?

File: packages/ui/src/components/ListV2/ListRow.tsx [hunk 1]
```diff
@@ -30,7 +34,7 @@ const StyledCheckbox = styled(Checkbox)`
   }
 `
 
-export const LIST_ROW_VARIANTS: Color[] = [
+export const LIST_ROW_SENTIMENTS: Color[] = [
```
Comment (matthprost) [inline packages/ui/src/components/ListV2/ListRow.tsx:37]: You can use `SENTIMENTS` variable from `packages/ui/src/theme/index.ts`

**Review by matthprost — CHANGES_REQUESTED** [review]

**Comment by matthprost** [comment]
There quite a lot of new code not covered by unit testing might be nice to update tests too 😄

File: packages/ui/src/components/ListV2/ListHeaderRow.tsx [hunk 2]
```diff
@@ -16,60 +16,43 @@ const StyledRow = styled('div', {
...
-      getSelectableRows().length > 0 &&
-      selectedIds.length === getSelectableRows().length
-    ) {
+    if (selectedIds.length === Object.keys(selectableIds).length) {
       return true
     }
 
-    return selectedIds.length > 0 ? 'indeterminate' : false
-  }, [getSelectableRows, selectedIds.length])
+    return selectedIds.length === 0 ? false : 'indeterminate'
```
[the remark this answers is not in the document]
Comment (fabienhebert) [reply packages/ui/src/components/ListV2/ListHeaderRow.tsx:46]: I get the point but It's sync with Checkbox `checked` prop, but keeping boolean for `checked` make sense. Ofc we can have a specific behavior here where a variable could be named `all | none | partial` and then convert this string value to the boolean or indeterminate but mb it sounds may be a bit overkill ? any opinion @matthprost ?

**Review by matthprost — APPROVED** [review]

File: packages/ui/src/components/ListV2/ListRow.tsx [hunk 3]
```diff
@@ -167,135 +166,121 @@ export const ListRow = ({
...
         data-highlight={!!isHighlighted || isSelected}
-        data-variant={variant}
-        role={isExpandable ? 'button row' : 'row'}
-        template={computedTemplate}
-        aria-expanded={isExpanded}
-        aria-haspopup={isExpandable}
-        onClick={isExpandable && !isDisabled ? handleExpand : undefined}
+        data-sentiment={sentiment}
+        role={expandable && isExpanded === undefined ? 'button row' : 'row'}
+        aria-expanded={isRowExpanded}
```
[the remark this answers is not in the document]
Comment (ModuloM) [reply packages/ui/src/components/ListV2/ListRow.tsx]: Still need something for this?
If not, you can resolve ;)

**Review by ModuloM — APPROVED** [review]

**Pull request closed without merging**